src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceVariableService.cs

```diff
@@ -182,29 +185,43 @@ private async Task CreateConstantFieldActionsAsync(State state, ArrayBuilder<Cod
                 // If something is a constant, and it doesn't access any other locals constants,
                 // then we prefer to offer to generate a constant field instead of a constant
                 // local.
-                var action1 = CreateAction(state, allOccurrences: false, isConstant: true, isLocal: false, isQueryLocal: false);
-                if (await CanGenerateIntoContainerAsync(state, action1, cancellationToken).ConfigureAwait(false))
-                {
-                    actions.Add(action1);
-                }
-
-                var action2 = CreateAction(state, allOccurrences: true, isConstant: true, isLocal: false, isQueryLocal: false);
-                if (await CanGenerateIntoContainerAsync(state, action2, cancellationToken).ConfigureAwait(false))
+                if (CanGenerateIntoContainer(state, cancellationToken))
                 {
-                    actions.Add(action2);
+                    actions.Add(CreateAction(state, allOccurrences: false, isConstant: true, isLocal: false, isQueryLocal: false));
+                    actions.Add(CreateAction(state, allOccurrences: true, isConstant: true, isLocal: false, isQueryLocal: false));
                 }
             }
         }
 
-        private async Task<bool> CanGenerateIntoContainerAsync(State state, CodeAction action, CancellationToken cancellationToken)
+        protected int GetFieldInsertionIndex(
+            bool isConstant, TTypeDeclarationSyntax oldType, TTypeDeclarationSyntax newType, CancellationToken cancellationToken)
         {
-            var result = await IntroduceFieldAsync(
-                state.Document, state.Expression,
-                allOccurrences: false, isConstant: state.IsConstant, cancellationToken: cancellationToken).ConfigureAwait(false);
+            var preferredInsertionIndex = isConstant
+                ? DetermineConstantInsertPosition(oldType, newType)
+                : DetermineFieldInsertPosition(oldType, newType);
+
+            var legalInsertionIndices = this.GetInsertionIndices(oldType, cancellationToken);
+            if (legalInsertionIndices[preferredInsertionIndex])
```

Comment:
We were checking if the index is out of range before, why is it removed?

Reply:
answered above.